nodecg@2.6.4 fails to start with Cannot find module '@effect/cluster/HttpRunner' when installed as bundle dependency

When using nodecg as a direct bundle dependency (as described in the experimental feature notice), npm start fails with a chain of missing @effect/* module errors on a fresh npm install.
Environment:

- NodeCG: 2.6.4
- Node.js: v22.22.2
- OS: Windows 11
- npm: latest

Steps to reproduce:

1. Install nodecg as a bundle dependency: "nodecg": "^2.6.4" in package.json
2. Run npm install
3. Run npx nodecg start

Expected behavior: NodeCG starts successfully.
Actual behavior: Fails with a chain of missing module errors:

1. Cannot find module '@effect/cluster/HttpRunner'
2. Cannot find module '@effect/rpc/RpcClient'
3. Cannot find module '@effect/sql/Migrator'
4. Cannot find module 'reflect-metadata'
5. Cannot find module 'ajv/dist/core'

Root cause: @effect/platform-node@0.100.0 (a dependency of nodecg) requires @effect/cluster, @effect/rpc, @effect/sql and other @effect/* packages, but these are not declared as dependencies in nodecg's own package.json. npm's deduplication does not install them in a way that nodecg can find them.
Workaround: Adding the missing packages as direct dependencies in the bundle's package.json resolves the issue:

> json"@effect/cluster": "0.52.11",
> "@effect/rpc": "0.72.2",
> "@effect/sql": "0.48.6",
> "@effect/workflow": "0.12.5",
> "@effect/experimental": "0.57.11",
> "reflect-metadata": "0.2.2",
> "ajv": "8.18.0",
> "ajv-draft-04": "1.0.0",
> "ajv-formats": "2.1.1"

Note: This issue did not occur prior to nodecg@2.6.x as earlier versions did not use the @effect ecosystem.

--- a/eslint.config.ts
+++ b/eslint.config.ts
@@ -31,6 +31,19 @@ export default defineConfig(
 			"@typescript-eslint/prefer-promise-reject-errors": "off",
 			"@typescript-eslint/unbound-method": "off",
 			"@typescript-eslint/prefer-nullish-coalescing": "off",
+
+			"@typescript-eslint/no-restricted-imports": [
+				"error",
+				{
+					paths: [
+						{
+							name: "@effect/platform-node",
+							message:
+								'Import a specific submodule (e.g. "@effect/platform-node/NodeRuntime")',
+						},
+					],
+				},
+			],
 		},
 		languageOptions: {
 			parserOptions: {

--- a/scripts/workspaces-parallel.ts
+++ b/scripts/workspaces-parallel.ts
@@ -1,6 +1,7 @@
 import * as cli from "@effect/cli";
 import { Command, FileSystem, Path } from "@effect/platform";
-import { NodeContext, NodeRuntime } from "@effect/platform-node";
+import NodeContext from "@effect/platform-node/NodeContext";
+import NodeRuntime from "@effect/platform-node/NodeRuntime";
 import { Array, Data, Effect, Fiber, Predicate, Schema } from "effect";
 
 const arg = cli.Args.text({ name: "command" });

--- a/workspaces/database-adapter-sqlite-legacy/package.json
+++ b/workspaces/database-adapter-sqlite-legacy/package.json
@@ -28,6 +28,7 @@
     "@nodecg/internal-util": "2.7.1",
     "@types/better-sqlite3": "^7.6.13",
     "better-sqlite3": "^12.4.1",
+    "reflect-metadata": "^0.2.2",
     "typeorm": "0.3.27"
   },
   "devDependencies": {

--- a/workspaces/nodecg/src/server/_effect/chokidar.test.ts
+++ b/workspaces/nodecg/src/server/_effect/chokidar.test.ts
@@ -1,7 +1,7 @@
 import path from "node:path";
 
 import { FileSystem } from "@effect/platform";
-import { NodeFileSystem } from "@effect/platform-node";
+import * as NodeFileSystem from "@effect/platform-node/NodeFileSystem";
 import { Chunk, Effect, Fiber, Stream } from "effect";
 import { describe, expect, test } from "vitest";
 

--- a/workspaces/nodecg/src/server/_effect/git-service.test.ts
+++ b/workspaces/nodecg/src/server/_effect/git-service.test.ts
@@ -1,7 +1,8 @@
 import * as path from "node:path";
 
 import { Command, FileSystem } from "@effect/platform";
-import { NodeContext, NodeFileSystem } from "@effect/platform-node";
+import * as NodeContext from "@effect/platform-node/NodeContext";
+import * as NodeFileSystem from "@effect/platform-node/NodeFileSystem";
 import { Chunk, Data, Effect, Layer, Option, Stream } from "effect";
 import { assert, describe, expect, test } from "vitest";
 

--- a/workspaces/nodecg/src/server/_effect/git-service.ts
+++ b/workspaces/nodecg/src/server/_effect/git-service.ts
@@ -1,7 +1,8 @@
 import nodeFs from "node:fs";
 
 import { FileSystem, Path } from "@effect/platform";
-import { NodeFileSystem, NodePath } from "@effect/platform-node";
+import * as NodeFileSystem from "@effect/platform-node/NodeFileSystem";
+import * as NodePath from "@effect/platform-node/NodePath";
 import { Array, Data, DateTime, Effect, Match, Option } from "effect";
 import git from "isomorphic-git";
 

--- a/workspaces/nodecg/src/server/bootstrap.ts
+++ b/workspaces/nodecg/src/server/bootstrap.ts
@@ -21,7 +21,7 @@ if (isLegacyProject()) {
 	}
 }
 
-import { NodeRuntime } from "@effect/platform-node";
+import * as NodeRuntime from "@effect/platform-node/NodeRuntime";
 import { ConfigError, Effect, Fiber } from "effect";
 
 import { UnknownError } from "./_effect/boundary";

